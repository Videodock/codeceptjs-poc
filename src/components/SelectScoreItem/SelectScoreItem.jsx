import React, { useState } from "react";
import * as PropTypes from 'prop-types';
import { Avatar, Box, Button, Card, IconButton, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/styles';
import InsertDriveFileIcon from '@material-ui/icons/InsertDriveFile';
import Launch from '@material-ui/icons/Launch';
import { useTranslation } from 'react-i18next';
import ScorePlaceholderIcon from "../Icons/ScorePlaceholderIcon";
import { truncateLabel } from "../../utils";
import styles from './SelectScoreItem.styles';

const useStyles = makeStyles(styles);

export default function SelectScoreItem({ item, isActiveCampaign, progress, onItemClick }){
  const { t }                   = useTranslation('selectComposition');
  const classes                 = useStyles();
  const [hover, setHover]       = useState(false);
  const [pdfHover, setPdfHover] = useState(false);
  const maxChars                = 200;

  return (
    <Card 
      className={classes.item} 
      elevation={hover? 4:2} 
      onMouseOver={() => setHover(true)}
      onMouseOut={() => setHover(false)}
      onClick={() => !pdfHover? onItemClick() : null}
    >
      <Box className={classes.imgBox}>
        <Avatar className={classes.img} src={item?.image} alt={item?.name || ""}>
          <ScorePlaceholderIcon />
        </Avatar>
      </Box>
      <Box className={classes.itemMain}>
        <Box className={classes.itemHeader}>
          <Typography variant="h3" color="primary">{item?.title || ""}</Typography>
          <Box className={classes.itemButtonBox}>
            {item?.format === "application/pdf" && 
              <a href={item?.source} target="_blank" rel="noopener noreferrer">
                <IconButton 
                  aria-label="PDF" 
                  onMouseOver={() => setPdfHover(true)}
                  onMouseOut={() => setPdfHover(false)} 
                >
                  <InsertDriveFileIcon className={classes.icon} />
                  <Typography>{t('pdf')}</Typography>
                </IconButton>
              </a>
            }
            {isActiveCampaign && 
              <a href={'<campaignlink>'} target="_blank" rel="noopener noreferrer">
                <Button 
                  variant="contained" 
                  size="small" 
                  color="primary"
                >
                  <Launch className={classes.icon} />
                  {t('active_campaign')}
                </Button>
              </a>
            }
          </Box>
        </Box>
        <Box>
          <Typography>{item?.subject}</Typography>
          <Typography className={classes.itemDescription} variant="body2">{truncateLabel(item?.description, maxChars)}</Typography>
          <Typography>
            {item?.creator && `${t('created_by')} ${item?.creator}`}
            {(item?.creator && item?.publisher) && ` â€¢ `}
            {item?.publisher && `${t('published_by')} ${item?.publisher}`}
          </Typography>
        </Box>
      </Box>
      {isActiveCampaign && 
        <Box className={classes.progressBar}>
          <Box className={classes.progress} style={{ width: `${progress}%` }}>&nbsp;</Box>
        </Box>
      }
    </Card>
  );
}

SelectScoreItem.propTypes = {
  item            : PropTypes.object,
  isActiveCampaign: PropTypes.bool,
  progress        : PropTypes.number,
  onItemClick     : PropTypes.func,
};

SelectScoreItem.defaultProps = {
  isActiveCampaign: false,
};