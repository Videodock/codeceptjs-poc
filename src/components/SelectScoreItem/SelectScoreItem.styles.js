import { createStyles } from '@material-ui/styles';

export default ({ spacing, palette, shape, breakpoints, typography }) => createStyles({
  item: {
    display      : 'flex',
    flexDirection: 'row',
    marginBottom : spacing(2),
    marginTop    : spacing(2),
    position     : 'relative',
    cursor       : 'pointer',
  },
  imgBox: {
    backgroundColor: palette.common.faintGrey,
    display        : 'flex',
    justifyContent : 'center',
    alignItems     : 'center',
    width          : 100,
  },
  img: {
    width          : '100%',
    height         : '100%',
    maxWidth       : 100,
    borderRadius   : 0,
    backgroundColor: palette.border.lightGrey,
    borderSize     : 3,
    color          : palette.shadow.transparentGrey,
    '& svg'        : {
      width : '2em',
      height: '2em',
    },
    '& img': {
      maxWidth : '90%',
      maxHeight: '90%',
    },
  },
  itemMain: {
    width  : '100%',
    height : '100%',
    padding: spacing(2),
  },
  itemHeader: {
    display       : 'flex',
    flexDirection : 'row',
    justifyContent: 'space-between',
  },
  itemButtonBox: {
    display      : 'flex',
    flexDirection: 'row',
    '& button'   : {
      borderRadius : 3,
      marginLeft   : spacing(2),
      paddingTop   : spacing(1),
      paddingBottom: spacing(1),
    },
    '& a': {
      textDecoration: 'none',
    },
  },
  itemDescription: {
    maxHeight   : 100,
    marginBottom: spacing(1),
  },
  iconButton: {
    borderRadius: spacing(1),
  },
  icon: {
    marginRight: spacing(1),
  },
  progressBar: {
    backgroundColor: palette.common.faintGrey,
    position       : 'absolute',
    bottom         : 0,
    width          : '100%',
    height         : 4,
  },
  progress: {
    backgroundColor: palette.primary.main,
  },
});
