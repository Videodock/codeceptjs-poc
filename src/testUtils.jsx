import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/styles';
import { I18nextProvider } from 'react-i18next';
import { ApolloProvider } from '@apollo/client';
import { BrowserRouter as Router } from 'react-router-dom';
import 'url-search-params-polyfill';
import i18n from '../src/i18n';
import client from './graphql';
import theme from './theme';
import store from './redux';

const wrapper = ({ children }) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>
          <Router>
            {children}
          </Router>
        </I18nextProvider>
      </ThemeProvider>
    </Provider>
  </ApolloProvider>
);

const customRender = (ui, options) => render(ui, { wrapper, ...options });

export { customRender as render };
