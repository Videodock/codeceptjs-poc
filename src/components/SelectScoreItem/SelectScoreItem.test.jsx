import React from 'react';
import { render } from '../../testUtils';
import SelectScoreItem from './SelectScoreItem';

describe('<SelectScoreItem />', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<SelectScoreItem />);

    expect(container.firstChild).toMatchSnapshot();
  });

  // add your unit tests here
});
