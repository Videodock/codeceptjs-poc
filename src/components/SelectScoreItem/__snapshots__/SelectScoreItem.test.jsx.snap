// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectScoreItem /> renders and matches snapshot 1`] = `
<div
  class="MuiPaper-root MuiCard-root makeStyles-item-1 MuiPaper-elevation2 MuiPaper-rounded"
>
  <div
    class="MuiBox-root MuiBox-root-40 makeStyles-itemImg-2"
  />
  <div
    class="MuiBox-root MuiBox-root-41 makeStyles-itemMain-3"
  >
    <div
      class="MuiBox-root MuiBox-root-42 makeStyles-itemHeader-4"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-colorPrimary"
      />
      <div
        class="MuiBox-root MuiBox-root-73 makeStyles-itemButtonBox-5"
      />
    </div>
    <div
      class="MuiBox-root MuiBox-root-74"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      />
      <p
        class="MuiTypography-root makeStyles-itemDescription-6 MuiTypography-body2"
      />
      <p
        class="MuiTypography-root MuiTypography-body1"
      />
    </div>
  </div>
</div>
`;
